forui/lib/src/widgets/progress.dart

```diff
@@ -0,0 +1,168 @@
+
+  @override
+  Widget build(BuildContext context) {
+    final style = this.style ?? context.theme.progressStyle;
+    final value = switch (this.value) {
+      < 0.0 => 0.0,
+      > 1.0 => 1.0,
+      _ => this.value,
+    };
+
+    return LayoutBuilder(
+      builder: (context, constraints) => ConstrainedBox(
+        constraints: style.constraints,
+        child: Stack(
+          alignment: AlignmentDirectional.centerStart,
+          children: [
+            Container(
+              decoration: style.backgroundDecoration,
+              width: constraints.maxWidth,
+            ),
+            AnimatedContainer(
+              curve: style.curve,
+              duration: style.animationDuration,
+              decoration: style.progressDecoration,
+              width: value * constraints.maxWidth,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DoubleProperty('value', value))
+      ..add(DiagnosticsProperty('style', style));
+  }
+}
+
+/// [FProgress]'s style.
+final class FProgressStyle with Diagnosticable {
+  /// The background's color.
+  final BoxDecoration backgroundDecoration;
+
+  /// The progress's color.
+  final BoxDecoration progressDecoration;
+
+  /// The constraints for the progress bar.
+  final BoxConstraints constraints;
+
+  /// The animation duration.
+  final Duration animationDuration;
+
+  /// The animation curve.
+  final Curve curve;
+
+  /// Creates a [FProgressStyle].
+  const FProgressStyle({
+    required this.backgroundDecoration,
+    required this.progressDecoration,
+    required this.constraints,
+    required this.animationDuration,
+    required this.curve,
+  });
+
+  /// Creates a [FProgressStyle] that inherits its properties from [colorScheme] and [style].
+  FProgressStyle.inherit({required FColorScheme colorScheme, required FStyle style})
+      : backgroundDecoration = BoxDecoration(
+          borderRadius: style.borderRadius,
+          color: colorScheme.secondary,
+        ),
+        progressDecoration = BoxDecoration(
+          borderRadius: style.borderRadius,
+          color: colorScheme.primary,
+        ),
+        constraints = const BoxConstraints(minHeight: 15.0, maxHeight: 15.0),
+        animationDuration = const Duration(milliseconds: 500),
+        curve = Curves.ease;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('progressDecoration', progressDecoration))
+      ..add(DiagnosticsProperty('backgroundDecoration', backgroundDecoration))
+      ..add(DiagnosticsProperty('constraints', constraints))
+      ..add(DiagnosticsProperty('animationDuration', animationDuration))
+      ..add(DiagnosticsProperty<Curve>('curve', curve));
```

Comment:
```suggestion
      ..add(DiagnosticsProperty('curve', curve));
```

The generics is implied.